tron/core/actionrun.py

```diff
@@ -616,13 +616,18 @@ def stop(self):
         if self.retries_remaining is not None:
             self.retries_remaining = -1
 
-        pass
+        self._kill_mesos_task()
 
     def kill(self, final=True):
         if self.retries_remaining is not None and final:
             self.retries_remaining = -1
 
-        pass
+        self._kill_mesos_task()
+        return "Warning: Mesos action might not be killed forcibly."
```

Comment:
I don't really understand, why wouldn't it? Mesos totally does a kill -9 if it doesn't die in like, 2 minutes or whatever.

Reply:
From https://github.com/Yelp/task_processing/blob/master/task_processing/plugins/mesos/logging_executor.py#L232, it looks like the kill function won't tell if it's a kill or a kill -9. Or am I missing anything?

Reply:
Its kinda deep into mesos land
http://mesos.apache.org/documentation/latest/configuration/agent/
executor_shutdown_grace_period (used to be docker_stop_timeout)
Really it is taken care of by docker, it will kill the container eventually, even if it has to do -9. We don't really get a choice.